doc: doxygen reports 5 errors from Algorithm and QRCodeEncoder at 5.x branch

### Describe the doc issue

With Ubuntu 24.04(x86-64) and doxygen 1.12.0, following errors are reported.
These errors (and warnings) are about javadoc.
I tried with 4.x branch, but there are no reports.

```
  [javadoc] Building index for all the packages and classes...
  [javadoc] Standard Doclet version 17.0.12+7-Ubuntu-1ubuntu224.04
  [javadoc] Building tree for all the packages and classes...
  [javadoc] /home/kmtr/work/build5-main/modules/java/jar/opencv/java/org/opencv/core/Algorithm.java:21: error: unknown tag: example
  [javadoc]  * @example samples/cpp/snippets/detect_blob.cpp
  [javadoc]    ^
  [javadoc] /home/kmtr/work/build5-main/modules/java/jar/opencv/java/org/opencv/objdetect/QRCodeEncoder.java:17: error: no tag name after @
  [javadoc]  * @{
  [javadoc]    ^
  [javadoc] /home/kmtr/work/build5-main/modules/java/jar/opencv/java/org/opencv/objdetect/QRCodeEncoder.java:26: error: no tag name after @
  [javadoc]  *     @{
  [javadoc]        ^
  [javadoc] /home/kmtr/work/build5-main/modules/java/jar/opencv/java/org/opencv/objdetect/QRCodeEncoder.java:43: error: no tag name after @
  [javadoc]  *     @}
  [javadoc]        ^
  [javadoc] /home/kmtr/work/build5-main/modules/java/jar/opencv/java/org/opencv/objdetect/QRCodeEncoder.java:45: error: no tag name after @
  [javadoc]  * @}
  [javadoc]
```

And it causes missing description of cv::Algorithm at 5.x branch.
- https://docs.opencv.org/4.x/d3/d46/classcv_1_1Algorithm.html
- https://docs.opencv.org/5.x/d3/d46/classcv_1_1Algorithm.html

### Fix suggestion

(1) it may be this `@example` should be independ other doxygen comments like others.

```bash
kmtr@kmtr-VMware-Virtual-Platform:~/work/opencv5$ grep "@example" modules/*/include/opencv2/*.hpp | head -20
modules/3d/include/opencv2/3d.hpp:/** @example samples/cpp/tutorial_code/features2D/Homography/pose_from_homography.cpp
modules/3d/include/opencv2/3d.hpp:/** @example samples/cpp/tutorial_code/features2D/Homography/homography_from_camera_displacement.cpp
modules/3d/include/opencv2/3d.hpp:/** @example samples/cpp/snippets/epipolar_lines.cpp
modules/3d/include/opencv2/3d.hpp:/** @example samples/cpp/tutorial_code/features2D/Homography/decompose_homography.cpp
modules/core/include/opencv2/core.hpp:/** @example samples/cpp/tutorial_code/ImgTrans/copyMakeBorder_demo.cpp
modules/core/include/opencv2/core.hpp:/** @example samples/cpp/tutorial_code/HighGUI/AddingImagesTrackbar.cpp
modules/core/include/opencv2/core.hpp:/** @example samples/cpp/tutorial_code/core/how_to_scan_images/how_to_scan_images.cpp
modules/core/include/opencv2/core.hpp:/** @example samples/python/snippets/dft.py
modules/core/include/opencv2/core.hpp:/** @example samples/cpp/pca.cpp
modules/core/include/opencv2/core.hpp:/** @example samples/cpp/tutorial_code/ml/introduction_to_pca/introduction_to_pca.cpp
modules/core/include/opencv2/core.hpp:/** @example samples/cpp/snippets/kmeans.cpp
modules/core/include/opencv2/core.hpp:/** @example samples/python/snippets/kmeans.py
modules/core/include/opencv2/core.hpp:@example samples/cpp/snippets/detect_blob.cpp <<---- HERE
modules/dnn/include/opencv2/dnn.hpp:/** @example samples/dnn/classification.cpp
modules/dnn/include/opencv2/dnn.hpp:/** @example samples/dnn/colorization.cpp
modules/dnn/include/opencv2/dnn.hpp:/** @example samples/dnn/object_detection.cpp
modules/dnn/include/opencv2/dnn.hpp:/** @example samples/dnn/openpose.cpp
modules/dnn/include/opencv2/dnn.hpp:/** @example samples/dnn/segmentation.cpp
modules/dnn/include/opencv2/dnn.hpp:/** @example samples/dnn/text_detection.cpp
modules/features2d/include/opencv2/features2d.hpp:/** @example samples/python/snippets/mser.py
```

(2) The root cause is lack of documentation for QR code encoder and others. 
It breaks QRCode* description of objdetect table 

https://docs.opencv.org/4.x/javadoc/org/opencv/objdetect/package-summary.html
> QRCodeEncoder	Groups the object candidate rectangles.

https://docs.opencv.org/5.x/javadoc/org/opencv/objdetect/package-summary.html
> QRCodeEncoder	objdetect Object Detection

--- a/modules/core/include/opencv2/core.hpp
+++ b/modules/core/include/opencv2/core.hpp
@@ -3114,6 +3114,10 @@ class CV_EXPORTS Algorithm;
 template<typename _Tp, typename _EnumTp = void> struct ParamType {};
 
 
+/** @example samples/cpp/snippets/detect_blob.cpp
+An example using the BLOB to detect and filter region.
+*/
+
 /** @brief This is a base class for all more or less complex algorithms in OpenCV
 
 especially for classes of algorithms, for which there can be multiple implementations. The examples
@@ -3124,9 +3128,6 @@ etc.).
 
 Here is example of SimpleBlobDetector use in your application via Algorithm interface:
 @snippet snippets/core_various.cpp Algorithm
-
-@example samples/cpp/snippets/detect_blob.cpp
-An example using the BLOB to detect and filter region.
 
 */
 class CV_EXPORTS_W Algorithm

--- a/modules/objdetect/include/opencv2/objdetect.hpp
+++ b/modules/objdetect/include/opencv2/objdetect.hpp
@@ -87,6 +87,7 @@ namespace cv
 //! @addtogroup objdetect_qrcode
 //! @{
 
+/** @brief QR code encoder. */
 class CV_EXPORTS_W QRCodeEncoder {
 protected:
     QRCodeEncoder();  // use ::create()
@@ -151,6 +152,8 @@ public:
     CV_WRAP virtual void encodeStructuredAppend(const String& encoded_info, OutputArrayOfArrays qrcodes) = 0;
 
 };
+
+/** @brief QR code detector. */
 class CV_EXPORTS_W_SIMPLE QRCodeDetector : public GraphicalCodeDetector
 {
 public:
@@ -192,6 +195,7 @@ public:
                                               OutputArray straight_qrcode = noArray());
 };
 
+/** @brief QR code detector with Aruco. */
 class CV_EXPORTS_W_SIMPLE QRCodeDetectorAruco : public GraphicalCodeDetector {
 public:
     CV_WRAP QRCodeDetectorAruco();

--- a/modules/objdetect/include/opencv2/objdetect.hpp
+++ b/modules/objdetect/include/opencv2/objdetect.hpp
@@ -195,7 +195,7 @@ public:
                                               OutputArray straight_qrcode = noArray());
 };
 
-/** @brief QR code detector with Aruco. */
+/** @brief QR code detector based on Aruco markers detection code. */
 class CV_EXPORTS_W_SIMPLE QRCodeDetectorAruco : public GraphicalCodeDetector {
 public:
     CV_WRAP QRCodeDetectorAruco();

